### PR TITLE
Make crate no_std

### PR DESCRIPTION
**Pull request by EHfive**
See commit messages.

The motivation is to use it for resizing emoji images in no_std cosmic-text, see https://github.com/pop-os/cosmic-text/pull/185.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name = "resize"
 version = "0.8.0"
 description = "Simple image resampling library in pure Rust."
 authors = ["Kornel <kornel@geekhood.net>", "Kagami Hiiragi <kagami@genshiken.org>"]
-categories = ["graphics", "multimedia::images"]
+categories = ["graphics", "multimedia::images", "no_std"]
```
Comment (kornelski) [inline Cargo.toml]: it's `no-std`

File: Cargo.toml [hunk 2]
```diff
@@ -15,13 +15,17 @@ edition = "2021"
 rust-version = "1.57"
 
 [features]
-default = ["rayon"]
+default = ["std", "rayon"]
+std = []
+no_std = ["libm", "hashbrown"]
```
Comment (kornelski) [inline Cargo.toml]: Please use `dep:libm`, `dep:hashbrown`, otherwise they become public features.

**Comment by kornelski** [comment]
Please don't use rustfmt. It makes code less readable by weirdly stretching it vertically. I deliberately don't use it.

File: Cargo.toml (hunk 2 above)
Comment (EHfive) [reply Cargo.toml]: https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies
> The `dep:` syntax is only available starting with Rust 1.60. Previous versions can only use the implicit feature name.

Bumped MSRV to 1.60.

**Pull request merged**